builtin/mainmenu/dlg_create_world.lua

```diff
@@ -105,38 +104,25 @@ local function create_world_formspec(dialogdata)
 			"button[4.75,2.5;3,0.5;world_create_cancel;" .. fgettext("Cancel") .. "]"
 	end
 
+	local current_mg = dialogdata.mg
 	local mapgens = core.get_mapgen_names()
 
-	local current_seed = core.settings:get("fixed_map_seed") or ""
-	local current_mg   = core.settings:get("mg_name")
 	local gameid = core.settings:get("menu_last_game")
 
-	local flags = {
-		main = core.settings:get_flags("mg_flags"),
-		v5 = core.settings:get_flags("mgv5_spflags"),
-		v6 = core.settings:get_flags("mgv6_spflags"),
-		v7 = core.settings:get_flags("mgv7_spflags"),
-		fractal = core.settings:get_flags("mgfractal_spflags"),
-		carpathian = core.settings:get_flags("mgcarpathian_spflags"),
-		valleys = core.settings:get_flags("mgvalleys_spflags"),
-		flat = core.settings:get_flags("mgflat_spflags"),
-	}
-
-	local gameidx = 0
-	if gameid ~= nil then
-		local _
-		_, gameidx = pkgmgr.find_by_gameid(gameid)
+	local flags = dialogdata.flags
 
-		if gameidx == nil then
-			gameidx = 0
-		end
+	local game, gameidx = pkgmgr.find_by_gameid(gameid)
+	if game == nil and hide_gamelist then
+		-- should never happen but just pick the first game
```

Comment:
This can happen if the user has selected a game that is now uninstalled
Also, there may be no games installed